src/viam/logging.py

```diff
@@ -1,11 +1,43 @@
+import asyncio
 import logging
 import sys
 from copy import copy
-from logging import DEBUG, ERROR, FATAL, INFO, WARN, WARNING  # noqa: F401
+from datetime import datetime
+from logging import DEBUG, ERROR, FATAL, INFO, WARN, WARNING, LogRecord  # noqa: F401
 from typing import Dict
 
+import viam
+
 LOG_LEVEL = INFO
 LOGGERS: Dict[str, logging.Logger] = {}
+_MODULE_PARENT = None
+
+
+class ModuleHandler(logging.Handler):
+    def __init__(self, logger: logging.Logger):
+        self.logger = logger
+        self.parent = _MODULE_PARENT
+
+        logger.handlers.clear()
+        super().__init__()
+
+    def emit(self, record: LogRecord):
+        assert isinstance(record, logging.LogRecord)
+        stack = f"exc_info: {record.exc_info}, exc_text: {record.exc_text}, stack_info: {record.stack_info}"
+        message = f"{record.filename}:{record.lineno}\t{record.msg}"
+        time = datetime.fromtimestamp(record.created)
+
+        try:
+            assert self.parent is not None
+            asyncio.create_task(
```

Comment:
I don't have a great sense of `asyncio` semantics. Is it possible to ever "leak" this task in the event that `self.parent.log` hangs forever?

Reply:
theoretically i think. we should be able to wrap it in a `wait_for`:

```python3
coro = asyncio.wait_for(self._parent.log(record.name, record.levelname, time, message, stack), 10)
asyncio.create_task(coro, name=f"{viam._TASK_PREFIX}-LOG-{record.created}")
```